src/hooks/useDrawerLlmChat.ts

```diff
@@ -0,0 +1,396 @@
+import { openai } from '@grafana/llm';
+import { useCallback, useRef, useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+
+import {
+  AttachedFile,
+  ChatMessage,
+  LlmHealthCheck,
+  LlmMessage,
+  UseChatMessagesReturn,
+  UseFileAttachmentsReturn,
+  UseLlmServiceReturn,
+  UseTextareaResizeReturn,
+} from '@/types';
+
+/**
+ * Configuration constants for the chat drawer
+ */
+export const chatConfig = {
+  maxFileSize: 10 * 1024 * 1024,
+  maxTextAreaHeight: 200,
+  minTextAreaHeight: 40,
+  requestTimeout: 15000,
+  temperature: 0.7,
+} as const;
+
+/**
+ * Allowed file types for attachment
+ */
+const allowedFileTypes: string[] = [
+  'image/jpeg',
+  'image/png',
+  'image/gif',
+  'image/webp',
+  'text/plain',
+  'application/pdf',
+  'application/msword',
+  'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+  'application/json',
+  'text/csv',
+] as const;
+
+/**
+ * Custom hook for managing chat messages
+ *
+ * Provides functionality for managing the chat message state including
+ * adding messages, updating messages, and generating unique IDs.
+ *
+ * @returns Object with message management functions
+ */
+export const useChatMessages = (): UseChatMessagesReturn => {
+  const [messages, setMessages] = useState<ChatMessage[]>([]);
+
+  /**
+   * Generates a unique message ID
+   * @returns Unique message identifier
+   */
+  const generateMessageId = useCallback((): string => {
+    return `msg-${uuidv4()}`;
+  }, []);
+
+  /**
+   * Adds a new message to the chat
+   * @param message - Message to add
+   */
+  const addMessage = useCallback((message: ChatMessage) => {
```

Comment:
I suppose `addMessages` can handle this case so the function is unnecessary and can be removed